src/code/z_collision_check.c

```diff
@@ -3147,17 +3185,17 @@ void func_80062E14(GlobalContext* globalCtx, Vec3f* v, Vec3f* arg2) {
         { 0, 0, 0, 0x00, 0x80, 0xFF, 0x00, 300 },
         0,
     };
-    s32 sp24;
+    s32 pIndex;
 
-    init.position.x = (s32)v->x;
-    init.position.y = (s32)v->y;
-    init.position.z = (s32)v->z;
+    init.position.x = v->x;
+    init.position.y = v->y;
+    init.position.z = v->z;
     init.lightPoint.x = init.position.x;
     init.lightPoint.y = init.position.y;
     init.lightPoint.z = init.position.z;
 
-    Effect_Add(globalCtx, &sp24, EFFECT_SHIELD_PARTICLE, 0, 1, &init);
-    Audio_PlaySoundGeneral(NA_SE_IT_REFLECTION_WOOD, arg2, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    Effect_Add(globalCtx, &pIndex, EFFECT_SHIELD_PARTICLE, 0, 1, &init);
+    Audio_PlaySoundGeneral(NA_SE_IT_REFLECTION_WOOD, actorPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
 }
 
 #define SQXZ(vec) (SQ(vec.x) + SQ(vec.z))
```

Comment:
perhaps something for math.h?

Reply:
I apparently hadn't merged this since the math3d cyl vs line seg update. this is in math.h now.